client/projector/Program.js

```diff
@@ -125,6 +126,11 @@ export default class Program extends React.Component {
         this._worker.postMessage({ messageId, receiveData: { object: this.props.papers } });
       } else if (sendData.name === 'markers') {
         this._worker.postMessage({ messageId, receiveData: { object: this.props.markers } });
+      } else if (sendData.name === 'camera') {
+        this._worker.postMessage({
+          messageId,
+          receiveData: { object: this.props.grabCameraImageAndProjectionData() },
+        });
```

Comment:
Might be more performant if we pass through an `ImageBitmap` as that can be sent efficiently to the Web Worker (see `transfer` in https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage). It would also simplify the code a bit because you can get an `ImageBitmap` directly from the video API without having to draw onto a canvas: https://w3c.github.io/mediacapture-image/#dom-imagecapture-grabframe

Reply:
See here for an example: https://simpl.info/imagecapture/

Reply:
awesome, thanks!